### PR TITLE
feat: add feedback feature to self-learning pages

### PR DESCRIPTION
**Pull request by gabriele-ct**
**Description of changes**
Adding feedback functionality connected to userguiding to self-learning pages

**Link to original ticket**
https://github.com/commercetools/commercetools-docs/issues/4637

**Link to change** (deep-link for reviewing the change)
https://docskit-ga-userguiding-script-injection.commercetools.vercel.app/self-learning-smoke-test/course-1/2-quiz

**Screenshot of changes (if applicable)**
Screenshot 2024-09-26 at 14 56 38

[DoD guidelines](https://commercetools.atlassian.net/wiki/spaces/ET/pages/738820530/Definition+of+Done)

- [ ] user documentation added?
- [x] stakeholders approve changes? (if needed)

### CODE REVIEW

**Comment by gabriele-ct** [comment]
@zbalek Would you mind having a look to what implemented so far: 
https://docskit-ga-userguiding-script-injection.commercetools.vercel.app/self-learning-smoke-test/course-1/2-quiz

Let me know if you have any observations or you'd like to change anything.
Thank you

**Comment by zbalek** [comment]
Looks great @gabriele-ct! The behavior is as expected.

Re the loading speed: Thanks for checking this! Imho, this is also really good. 

Re mobile: I cross-checked the requirements and we actually haven't defined the situation for mobile _(which is when users may be the most likely to access the pages using 4G)_. And so I was wondering if it is possible to only enable the feedback feature only up to a tablet size screen - at least for the MVP. _(I tried to test the mobile version (SE) just now and it is breaking a bit.)_

------------

Can we please increase the margin here from 16 to 24:
image

**Comment by gabriele-ct** [comment]
@zbalek could you please have another look? I addressed the points you suggested. Thank you

**Comment by zbalek** [comment]
> @zbalek could you please have another look? I addressed the points you suggested. Thank you

All good now - thank you!

**Comment by gabriele-ct** [comment]
@FFawzy I haven't been able to reproduce the non loading survey issue, would you mind double check if we're good to merge this?

**Comment by FFawzy** [comment]
@gabriele-ct .. Unfortunately I can still reproduce it. 
it happens for fresh new customers please check the video attached
one more thing, I can't find the [google tracker](https://github.com/commercetools/commercetools-docs/issues/4637#issuecomment-2360322621) on the thumbs events did I miss it ?

https://github.com/user-attachments/assets/31a7d5ca-00bb-40d1-b977-c36af6491a82

**Comment by gabriele-ct** [comment]
Thank you for finding out the repro path. It helped a lot. 
Turns out that when the script loads it creates some items in  localstorage, one of them looks like the session id, nothing happens until that is present in the localstorage. I'm therefore waiting for that session id to exists before calling any library method. It seems to be working all the times.

I also added google tag calls. It should be ready for a final review 🙇

**Comment by FFawzy** [comment]
Pinging @nkuehn as he is defacto the Self Learning stakeholder for this feature. 
can you please test out the feature here
https://docskit-ga-userguiding-script-injection.commercetools.vercel.app/self-learning-smoke-test/course-1/2-quiz

Thank you

**Comment by nkuehn** [comment]
Oh, thanks!  I like it a lot, very nice direct user experience. 

I looked at the debug / log a bit, too and it seems we're really only loading userguiding when needed.  

Did you check with data protection team whether we they have to do something that the cookie consent banner is maybe re-shown again (or, at least that they have documented that we have userguiding on the docs site now)?

**Comment by FFawzy** [comment]
> Oh, thanks! I like it a lot, very nice direct user experience.
> I looked at the debug / log a bit, too and it seems we're really only loading userguiding when needed.

Properly specified ✅ 
properly implemented ✅ 
🎉 

> Did you check with data protection team whether we they have to do something that the cookie consent banner is maybe re-shown again (or, at least that they have documented that we have userguiding on the docs site now)?

I'll send them an email to check

**Review by ColinRosati — COMMENTED** [review]
Looking good. Just a few small comments

File: packages/gatsby-theme-docs/src/components/page-feedback.tsx [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+const PageFeedbackWrapper = styled.div`
+  border-top: 1px solid ${designSystem.colors.light.borderPrimary};
+  padding-top: ${designSystem.dimensions.spacings.l};
+  font-size: ${designSystem.typography.fontSizes.small};
+`;
+
+const PageFeedback = () => {
+  const [currentFeedback, setCurrentFeedback] = useState(0);
+  const isScriptLoaded = (): boolean => {
+    const isUserGuidingSessionReady =
```
[the remark this answers is not in the document]
Comment (gabriele-ct) [reply packages/gatsby-theme-docs/src/components/page-feedback.tsx:31]: These functions both refers to objects not available in SSR, so I'd keep them here to have the guarantee they don't break during SSR.

[the remark this answers is not in the document]
Comment (ColinRosati) [reply packages/gatsby-theme-docs/src/components/page-feedback.tsx:31]: It would be interesting to test that theory out 🤔

**Review by ColinRosati — APPROVED** [review]
LGTM

File: packages/gatsby-theme-docs/src/components/page-feedback.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (gabriele-ct) [reply packages/gatsby-theme-docs/src/components/page-feedback.tsx:31]: not a theory, you can go ahead and test it locally

**Comment by gabriele-ct** [comment]
@FFawzy shall we consider this PR "blocked" until we hear back from data protection team?

**Comment by FFawzy** [comment]
@gabriele-ct we are good to go 🙌 

they only needed me to file a report with some details and I did it so we are good to go.

**Pull request merged**